clients/drcachesim/common/options.cpp

```diff
@@ -1064,5 +1064,14 @@ droption_t<bool> op_abort_on_invariant_error(
     "total invariant error count is printed at the end; a non-zero error count does not "
     "affect the exit code of the analyzer.");
 
+droption_t<bool> op_pt2ir_best_effort(
+    DROPTION_SCOPE_ALL, "pt2ir_best_effort", false,
+    "Whether errors encountered during PT trace conversion in pt2ir are ignored.",
+    "By default, errors in decoding the kernel syscall PT trace in pt2ir are fatal to "
+    "raw2trace. With -pt2ir_best_effort, those errors do not cause failures and their "
+    "counts are reported by raw2trace at the end. This may result in a trace where not "
+    "all syscalls have a trace, and the ones that do may have some PC discontinuities. "
+    "These discontinuities will still be reported by the invariant checker.");
```

Comment:
What about the missing syscall traces: also reported by the invariant checker?

Reply:
No that won't be. There are some syscalls that we don't trace intentionally (exit, sigreturn: https://github.com/DynamoRIO/dynamorio/blob/d2eb7ec343abd6e5b483faceb06425bf5297d148/clients/drcachesim/tracer/syscall_pt_trace.cpp#L287). But it could. Added a TODO.

Reply:
So if the user turns on -pt2ir_best_effort it may not be easy to find out how many syscalls ended up empty and omitted, or whether any where.  They can look at the raw2trace stats during conversion, yes.  But the process makes real trace files, and there's no indication let's say half the syscalls are empty: so if they or someone else later (maybe forgetting about the raw2trace stats; are they stored?) runs some analyzers on them they might get surprising results and not understand why.

Reply:
I would say we should add to the option docs here that the user needs to carefully look at the empty syscall stats, as there's no invariant check.

Reply:
Listed out the stats the user should look at